### DISCUSSION TITLE
chatClient.closeConnection() and chatClient.disconnectUser()

### DISCUSSION THREAD

**Discussion by Abacaxi-Nelson** [GetStream/stream-chat-flutter#677] [Q&A] [answered]
Hi all

im lost between 
```dart
chatClient.closeConnection();
await chatClient.disconnectUser();
```

Do we need to call both on signout ?

**Comment by imtoori** [comment, marked as answer]
hey @Abacaxi-Nelson 
`closeConnection` closes the connection without clearing the user
This will not trigger default auto-retry mechanism for reconnection.
You need to call `openConnection` to reconnect.

`disconnectUser` closes the connection and clears the state of the client

you would call the former when you want to temporarily close the connection, the latter is used for signout
